azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -230,6 +240,11 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+      // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
+      // if the directory is not permissioned correctly already (should happen once per execution)
+      if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
```

Comment:
If you plan to address it soon, documenting it here is easier. Otherwise, I will leave it to you. Just consider how others can find and use this information to make documenting it worthwhile.